python/pyspark/sql/udf.py

```diff
@@ -295,6 +311,20 @@ def returnType(self) -> DataType:
                         "return_type": str(self._returnType_placeholder),
                     },
                 )
+        elif self.evalType == PythonEvalType.SQL_COGROUPED_MAP_ARROW_UDF:
+            if isinstance(self._returnType_placeholder, StructType):
+                try:
+                    to_arrow_type(self._returnType_placeholder)
+                except TypeError:
+                    raise NotImplementedError(
```

Comment:
Shall we raise `PySparkNotImplementedError` instead?

Reply:
done